wp-content/plugins/hurumap/src/FlourishBlock/Chart.js

```diff
@@ -3,49 +3,85 @@ import React from 'react';
 
 import InsightContainer from '@codeforafrica/hurumap-ui/core/InsightContainer';
 import makeStyles from '@material-ui/styles/makeStyles';
+import ArrowDropUp from '@material-ui/icons/ArrowDropUp';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
 import propTypes from '../propTypes';
 
-const useStyles = makeStyles({
+const useStyles = makeStyles(({ palette, breakpoints }) => ({
   root: {
     width: '100%',
     padding: '1.25rem'
   },
-  iframe: {}
-});
+  iframe: {},
+  descriptionWrapper: {
+    margin: '1.25rem auto 0',
+    [breakpoints.up('md')]: {
+      width: '90%'
+    }
+  },
+  description: {
+    color: palette.data.main,
+    display: 'flex',
+    marginLeft: '1.25rem'
+  }
+}));
 
-function Chart({ title, chartId, iframeKey, ...props }) {
+function Chart({ title, description, chartId, iframeKey, ...props }) {
   const classes = useStyles(props);
   return (
-    <InsightContainer
-      hideInsight
-      key={chartId}
-      variant="analysis"
-      loading={false}
-      title={title}
-    >
-      <div />
-      <div className={classes.root}>
-        <iframe
-          key={iframeKey}
-          frameBorder="0"
-          scrolling="no"
-          title={title}
-          src={`/wp-json/hurumap-data/flourish/${chartId}/`}
-          className={classes.iframe}
-        />
-      </div>
-    </InsightContainer>
+    <>
+      <InsightContainer
+        hideInsight
+        key={chartId}
+        variant="analysis"
+        loading={false}
+        title={title}
+        embedCode="embed text"
+        actions={{
+          handleShare: () => {}
+        }}
+      >
+        <div />
+        <div className={classes.root}>
+          <iframe
+            key={iframeKey}
+            frameBorder="0"
+            scrolling="no"
+            title={title}
+            src={`/wp-json/hurumap-data/flourish/${chartId}/`}
+            className={classes.iframe}
+          />
+        </div>
+      </InsightContainer>
+      {description && (
+        <div className={classes.descriptionWrapper}>
+          <Grid container alignItems="flex-start" wrap="nowrap">
+            <Grid item>
+              <ArrowDropUp color="primary" />
+            </Grid>
+            <Grid item>
+              <Typography variant="caption" className={classes.description}>
+                {description}
+              </Typography>
+            </Grid>
+          </Grid>
+        </div>
+      )}
+    </>
```

Comment:
Lets add a story to move `description` into chart containers themselves (so they can be part of embed/share/download actions)

Reply:
@kilemensi got you! :+1: